estate/models/estate_property.py

```diff
@@ -0,0 +1,111 @@
+    postcode = fields.Char()
+    date_availability = fields.Date(copy=False, default=lambda self: date.today() + relativedelta(months=3))
+    expected_price = fields.Float(required=True)
+    selling_price = fields.Float(readonly=True, copy=False)
+    bedrooms = fields.Integer(default=2)
+    living_area = fields.Integer()
+    facade = fields.Integer()
+    garage = fields.Boolean()
+    garden = fields.Boolean()
+    garden_area = fields.Integer()
+    garden_orientation = fields.Selection(
+        string='Garden Orientation',
+        selection=[
+            ('north','North'),
+            ('south','South'), 
+            ('east','East'), 
+            ('west','West')
+        ],
+    )
+    # active = fields.Boolean(default=False)
+    estate = fields.Selection(
+        string = 'Estado',
+        selection=[('new','New'),('offer received','Offer Received'),('offer accepted','Offer Accepted'),('sold','Sold'),('canceled','Canceled')],
+        required=True,
+        copy=False,
+        default='new'
+    )
+
+
+    # relation between models
+    property_type_id = fields.Many2one("estate.property.type", string="Property type" )
+
+    buyer = fields.Many2one("res.partner", string="Buyer", copy=False)
+    
+    salesperson = fields.Many2one("res.users",string="Sales man", default=lambda self: self.env.user)
+
+    tags_ids = fields.Many2many("estate.property.tag",string="Tags")
+
+    offer_ids = fields.One2many("estate.property.offer","property_id",string="Offers")
+   
+
+    total_area = fields.Float(compute="_compute_total_area")
+    best_price = fields.Float(compute="_compute_maximum_price")
+
+
+
+    @api.depends("garden_area","living_area")
+    def _compute_total_area(self):
+        for record in self:
+            record.total_area = record.garden_area + record.living_area
+
+    @api.depends("offer_ids.price")
+    def _compute_maximum_price(self):
+        for record in self:
+            offer_prices = record.offer_ids.mapped('price')
+            record.best_price = max(offer_prices) if offer_prices else 0.0
+            
+
+    @api.onchange("garden")
+    def _onchange_garden(self):
+        if self.garden:
+            self.garden_area = 10
+            self.garden_orientation = "north"
+        else:
+            self.garden_area = None
+            self.garden_orientation = None
+
+
+    def action_sold(self):
+        for record in self:
+            if record.estate == "canceled":
+                raise UserError("Canceled property cannot be sold")
+            record.estate = "sold"
+            return True
+
+    def action_cancel(self):
+        for record in self:
+            if record.estate == "sold":
+                raise UserError("Sold property cannot be cancel")
+            record.estate = "canceled"
+            return True
+
+
+    _sql_constraints = [
+            ('check_expected_price', 'CHECK(expected_price >= 0)',
+            'The expected_price of should be greater than 0.'),
```

Comment:
This is a message that can be shown to the user (if I remember correctly), we shouldn't be using technical terms

```suggestion
            'The expected price should be greater than 0.'),
```